plugins/com.yakindu.solidity.compiler/src/com/yakindu/solidity/compiler/builder/ISolidityCompiler.java

```diff
@@ -19,11 +19,14 @@
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.emf.common.util.URI;
 
+import com.google.inject.ImplementedBy;
+
 /**
  * 
  * @author Florian Antony - Initial contribution and API
  *
  */
+@ImplementedBy(SolidityCompilerBase.class)
```

Comment:
If we do not want to use default implementations, we should consider adding a macos fragment that contains a simple SolidityCompiler implementation which just checks preferences for a compiler path.